From ebeda39b1b621c9443e1d93e6a36dac79d18ffa3 Mon Sep 17 00:00:00 2001
From: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Date: Wed, 5 Jul 2017 13:28:24 -0400
Subject: [PATCH 2591/2831] drm/amd/display: Undo dc_update_surfaces_and_stream
 change.

Restoring conditional call to in_transfer_func since no reason
to call it unconditionally.

Change-Id: I1b83df8e06f0e7bdb2688282310361993458a0a7
Signed-off-by: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Reviewed-by: Tony Cheng <Tony.Cheng@amd.com>
Acked-by: Harry Wentland <Harry.Wentland@amd.com>
---
 drivers/gpu/drm/amd/display/dc/core/dc.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/core/dc.c b/drivers/gpu/drm/amd/display/dc/core/dc.c
index 3a0606a..b2e9405 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc.c
@@ -1603,10 +1603,7 @@ void dc_update_surfaces_and_stream(struct dc *dc,
 			if (update_type == UPDATE_TYPE_FAST)
 				continue;
 
-			/* TODO find out why check is false */
-			/* TODO with this still not programming some color stuff... panel is dark-ish */
-			/*if (is_new_pipe_surface ||
-					srf_updates[i].in_transfer_func)*/
+			if (srf_updates[i].in_transfer_func)
 				core_dc->hwss.set_input_transfer_func(
 						pipe_ctx, pipe_ctx->surface);
 
-- 
2.7.4

